### PR TITLE
Service create

### PR DESCRIPTION
**Pull request by lienht-1767**
==> T
==> Redmine: https://edu-redmine.sun-asterisk.vn/issues/8970
==> Attachment File:
Screenshot from 2019-03-30 22-09-43
Screenshot from 2019-03-30 22-10-47

Screenshot from 2019-04-01 08-38-48

### CODE REVIEW

**Comment by lienht-1767** [comment]
ready

File: app/models/service.rb [hunk 1]
```diff
@@ -3,9 +3,22 @@ class Service < ApplicationRecord
...
+  validates :price, presence: true, numericality: {only_float: true}
+  validates :status, presence: true
+  mount_uploader :picture, PictureUploader
 
   scope :all_services, ->{select :id, :name, :description, :price}
   scope :other_services, ->(id){where.not id: id}
   scope :by_lastest, -> {order created_at: :desc}
+
+  def picture_size
+    if picture.size > Settings.size_picture.megabytes
```
Comment (huuphuc93) [inline app/models/service.rb]: return  unless

File: app/views/admin/services/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<% provide :title, t(".title") %>
+<div class="col-md-8">
+  <%= form_for [:admin, @service], class: "nifty_form" do |f| %>
+    <%= render "shared/error_messages", object: f.object %>
+    <%= f.label :name %>
+    <%= f.text_field :name %>
+    <%= f.label :status %>
+    <%= f.select :status, options_for_select([t(".public"), t(".pending")],
+      {:selected => @service.status}) %>
```
Comment (huuphuc93) [inline app/views/admin/services/new.html.erb]:
```suggestion
      selected: @service.status) %>
```

File: app/views/admin/services/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    <%= f.label :picture %>
+    <%= f.file_field :picture, accept: "image/jpeg,image/gif,image/png",
+          class: "picture" %>
```
Comment (huuphuc93) [inline app/views/admin/services/new.html.erb]: indent 2 spaces thôi em

File: app/views/admin/services/edit.html.erb [hunk 4]
```diff
@@ -1,13 +1,19 @@
...
-  <%= f.text_field :price %>
-  <%= f.text_field :description %>
-  <%= f.submit t(".update"), class: "col-md-3 btn btn-success" %>
-<% end %>
+    <%= render "shared/error_messages", object: f.object %>
+    <%= f.label :name %>
+    <%= f.text_field :name %>
+    <%= f.label :status %>
+    <%= f.select :status, options_for_select([t(".public"), t(".pending")],
+      {:selected => @service.status}) %>
```
Comment (huuphuc93) [inline app/views/admin/services/edit.html.erb]:
```suggestion
          selected: @service.status) %>
```

File: app/views/admin/services/edit.html.erb [hunk 5]
```diff
@@ -1,13 +1,19 @@
...
+    <%= f.label :price %>
+    <%= f.text_field :price %>
+    <%= f.label :picture %>
+    <%= f.file_field :picture, accept: "image/jpeg,image/gif,image/png",
+          class: "picture" %>
```
Comment (huuphuc93) [inline app/views/admin/services/edit.html.erb]: indent

**Comment by lienht-1767** [comment]
ready

File: app/models/service.rb [hunk 6]
```diff
@@ -3,9 +3,21 @@ class Service < ApplicationRecord
...
+  validates :status, presence: true
+  mount_uploader :picture, PictureUploader
 
   scope :all_services, ->{select :id, :name, :description, :price}
   scope :other_services, ->(id){where.not id: id}
   scope :by_lastest, -> {order created_at: :desc}
+
+  def picture_size
+    return unless picture.size > Settings.size_picture.megabytes
+      errors.add(:picture, I18n.t(".should_be_less_than_5MB"))
```
Comment (huuphuc93) [inline app/models/service.rb]: indent

File: app/views/admin/services/edit.html.erb [hunk 7]
```diff
@@ -1,13 +1,19 @@
...
-    {:selected => @service.status}) %>
-  <%= f.text_field :price %>
-  <%= f.text_field :description %>
-  <%= f.submit t(".update"), class: "col-md-3 btn btn-success" %>
-<% end %>
+    <%= render "shared/error_messages", object: f.object %>
+    <%= f.label :name %>
+    <%= f.text_field :name %>
+    <%= f.label :status %>
+    <%= f.select :status, options_for_select([t(".public"), t(".pending")],
```
Comment (huongnguyenmta) [inline app/views/admin/services/edit.html.erb]: viết method helper để tái sử dụng em nhé `[t(".public"), t(".pending")]`

**Review by huongnguyenmta — COMMENTED** [review]
Nhân tiện update lại status ticket em nhé
Gửi pull rồi thì status = Reviewing nhé :D

File: app/views/admin/services/edit.html.erb (hunk 7 above)
Comment (lienht-1767) [reply app/views/admin/services/edit.html.erb]: e update rồi ạ :D

**Comment by lienht-1767** [comment]
ready

**Comment by huuphuc93** [comment]
commented
gộp commit nhé

**Comment by lienht-1767** [comment]
ready

**Pull request merged**